lib/runtime/src/transports/etcd.rs

```diff
@@ -311,105 +314,189 @@ impl Client {
         prefix: impl AsRef<str> + std::fmt::Display,
         include_existing: bool,
     ) -> Result<PrefixWatcher> {
-        let client = self.connector.get_client();
-        let mut kv_client = client.kv_client();
-        let mut watch_client = client.watch_client();
+        let (tx, rx) = mpsc::channel(32);
+
+        // Get start revision and send existing KVs
+        let mut start_revision = self
+            .get_start_revision(
+                prefix.as_ref(),
+                if include_existing { Some(&tx) } else { None },
+            )
+            .await?;
+
+        // Resilience watch stream in background
+        let connector = self.connector.clone();
+        let prefix_str = prefix.as_ref().to_string();
+        self.rt.spawn(async move {
+            let mut reconnect = true;
+            while reconnect {
+                // Start a new watch stream
+                let watch_stream =
+                    match Self::new_watch_stream(&connector, &prefix_str, start_revision).await {
+                        Ok(stream) => stream,
+                        Err(_) => return,
+                    };
+
+                // Watch the stream
+                reconnect =
+                    Self::monitor_watch_stream(watch_stream, &prefix_str, &mut start_revision, &tx)
```

Comment:
I see a few error spots that `return true` such that `reconnect=true` for the while loop to trigger starting a new stream. But I also saw some error spots that `return false`, so I wasn't quite sure which errors require a reconnect and which ones do not at a glance. Can you add a docstring to `async fn monitor_watch_stream` that explains it a bit more in detail at a high level?

Reply:
Sure, added docstring to helper methods/functions
[docs: Comments on etcd helper methods/functions](https://github.com/ai-dynamo/dynamo/pull/3950/commits/fea73c0c4aa82e2e6dcbdd980173f674500be937)